http_service/models.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import lzma
+import os
+import shutil
+from urllib.request import urlretrieve
+
+import requests
+
+from bugbug.models.component import ComponentModel
+from bugbug.models.defect_enhancement_task import DefectEnhancementTaskModel
+from bugbug.models.regression import RegressionModel
+
+MODELS = {
+    "defectenhancementtask": DefectEnhancementTaskModel,
+    "component": ComponentModel,
+    "regression": RegressionModel,
+}
+
+
+def retrieve_model(name):
+    os.makedirs("models", exist_ok=True)
+
+    file_name = f"{name}model"  # noqa: E999
+    file_path = os.path.join("models", file_name)
+
+    model_url = f"https://index.taskcluster.net/v1/task/project.releng.services.project.testing.bugbug_train.latest/artifacts/public/{file_name}.xz"  # noqa
+    print(f"Checking ETAG of {model_url}")
+    r = requests.head(model_url, allow_redirects=True)
+    new_etag = r.headers["ETag"]
+
+    try:
+        with open(f"{file_path}.etag", "r") as f:  # noqa
+            old_etag = f.read()
+    except IOError:
+        old_etag = None
+
+    if old_etag != new_etag:
+        print(f"Downloading the model from {model_url}")
+        urlretrieve(model_url, f"{file_path}.xz")
+
+        with lzma.open(f"{file_path}.xz", "rb") as input_f:  # noqa
+            with open(file_path, "wb") as output_f:
+                shutil.copyfileobj(input_f, output_f)
+
+        with open(f"{file_path}.etag", "w") as f:  # noqa
+            f.write(new_etag)
+    else:
+        print(f"ETAG for {model_url} is ok")
+
+    return file_path
+
+
+def get_model_path(name):
+    file_name = f"{name}model"  # noqa: E999
```

Comment:
Nit: can you make this black-compliant and remove noqa?

Reply:
It should be black-compliant, I removed the noqa